igvc_navigation/src/back_circle/back_circle_server.cpp

```diff
@@ -0,0 +1,82 @@
+#include "back_circle_server.h"
+
+BackCircleServer::BackCircleServer()
+{
+  back_circle_service_server_ =
+      nh_.advertiseService("/back_circle_service", &BackCircleServer::backCircleCallback, this);
+  ROS_INFO("Back Circle Service Ready");
+}
+
+void BackCircleServer::waitForTransform()
+{
+  const double waiting_time = 5.0;
+  while (!tf_listener_.waitForTransform("odom", "base_footprint", ros::Time::now(), ros::Duration(waiting_time)))
+  {
+    ROS_INFO_STREAM("base_footprint->odom transform not found. waiting...");
+  }
+  ROS_INFO_STREAM("base_footprint->odom transform found!");
+}
+
+bool BackCircleServer::backCircleCallback(igvc_msgs::BackCircle::Request &req, igvc_msgs::BackCircle::Response &res)
+{
+  ros::NodeHandle pNh("~");
+
+  double width, length, grid_size, thickness, offset;
+  std::vector<double> xVec, yVec, zVec;
+
+  assertions::getParam(pNh, std::string("width"), width);
+  assertions::getParam(pNh, std::string("length"), length);
+  assertions::getParam(pNh, std::string("grid_size"), grid_size);
+  assertions::getParam(pNh, std::string("thickness"), thickness);
+  assertions::getParam(pNh, std::string("offset"), offset);
+  double meters = (length + width) * 2;
+
+  int img_size = static_cast<int>(std::round(meters / grid_size));
+  cv::Mat img(img_size, img_size, CV_8U, cv::Scalar::all(0));
+  cv::ellipse(img, cv::Point(img_size / 2, img_size / 2), cv::Size(width / grid_size, length / grid_size), 0, 90, 270,
+              cv::Scalar(255), static_cast<int>(std::round(thickness / grid_size)));
+
+  pcl::PointCloud<pcl::PointXYZ> back_circle_pointcloud_base_footprint;
+  back_circle_pointcloud_base_footprint.header.stamp = ros::Time::now().toNSec() / 1000;
+  back_circle_pointcloud_base_footprint.header.frame_id = "/base_footprint";
```

Comment:
Why is does a slash prefix `base_footprint` here but elsewhere there no slashes?

Reply:
It doesn't really matter, and looking at the file it's in it also uses the slash prefix.